BookTracking.Web/src/pages/Home/BookTable.tsx

```diff
@@ -0,0 +1,119 @@
+import { Fragment, ReactNode, useState } from 'react';
+import { Anchor, Badge, Button, Collapse, Flex, Image, Table } from '@mantine/core';
+import { useDisclosure } from '@mantine/hooks';
+import { Book } from '../../models/Book';
+import { ExpandedBookRow } from './ExpandedBookRow';
+
+import './BookTable.css';
+
+const rowData: Book[] = [
+  {
+    id: '1',
+    authorId: '1',
+    link: new URL('https://www.google.com'),
+    coverImage: './src/assets/testImage.webp',
+    title: 'Book Title',
+    author: 'Author',
+    tags: ['Fantasy', 'Magic'],
+    rating: 7,
+  },
+  {
+    id: '2',
+    authorId: '2',
+    link: new URL('https://www.google.com'),
+    coverImage: './src/assets/testImage.webp',
+    title: 'Book Title 2',
+    author: 'Author',
+    tags: ['Fantasy', 'Magic'],
+    rating: 8,
+  },
+  {
+    id: '3',
+    authorId: '3',
+    link: new URL('https://www.google.com'),
+    coverImage: './src/assets/testImage.webp',
+    title: 'Book Title 3',
+    author: 'Author',
+    tags: ['Fantasy', 'Magic', 'Sci-Fi', 'History', 'Progression', 'Cultivation', 'Fantasy'],
+    rating: 9,
+  },
+  {
+    id: '4',
+    authorId: '4',
+    link: new URL('https://www.google.com'),
+    coverImage: './src/assets/testImage.webp',
+    title: 'Book Title 4',
+    author: 'Author',
+    tags: ['Fantasy', 'Magic'],
+    rating: 10,
+    lastReadBookOrChapter: 'Ch. 23',
+  },
+];
+
+export const BookTable = () => {
+  const [openedRows, setOpenedRows] = useState<number[]>([]);
+
+  const toggleOpened = (index: number) => {
+    setOpenedRows((prev) =>
+      prev.includes(index) ? prev.filter((i) => i !== index) : [...prev, index]
+    );
+  };
+
+  const rows = rowData.map((book, index) => (
+    <>
+      <Table.Tr key={book.id}>
+        <Table.Td>
+          <Image radius="md" src={book.coverImage} h={120} fit="contain" />
+        </Table.Td>
+        <Table.Td>
+          <Anchor href={book.link.href} target="_blank">
+            {book.title}
+          </Anchor>
+        </Table.Td>
+        <Table.Td>{book.author}</Table.Td>
+        <Table.Td>
+          <Flex wrap="wrap" gap="sm">
+            {book.tags.map((tag, index) => (
+              <Badge key={index} color="blue" variant="filled">
```

Comment:
Similar to other keys, be aware if this mapping is dynamic and the elements can change either from external data, adding, removing, or sorting, the `key` should not be something determined at render time, ie `index`. In this case I suspect these tags won't change so it is fine, but something to note.